### PR TITLE
Fix signature of used `protocol_confirmation`/`analyzer_confirmation`.

### LINKED ISSUE #8: Error with Zeek v4.1.1
https://github.com/zeek/spicy-ldap/blob/70b186e926a19501122fb7c616dc5008c195ac87/analyzer/main.zeek#L292

I'm running Zeek v4.1.1 and this error comes up:

```
logger: error in [REDACTED]/packages/spicy-ldap/main.zeek, line 292: identifier not defined: AllAnalyzers::Tag
logger: error in [REDACTED]/packages/spicy-ldap/main.zeek, line 292 and event.bif.zeek, line 406: use of undeclared alternate prototype (event(c:connection; atype:error; aid:count;) and protocol_confirmation)
```

### PR DESCRIPTION
**Pull request by bbannier**
Closes #8.

### CODE REVIEW

File: analyzer/main.zeek [hunk 1]
```diff
@@ -289,7 +289,7 @@ function set_session(c: connection, message_id: int, opcode: LDAP::ProtocolOpcod
 @if (Version::at_least("4.2.0"))
 event analyzer_confirmation(c: connection, atype: Analyzer::Tag, aid: count) {
```
Comment (rsmmr) [inline analyzer/main.zeek]: Actually I think *this* should be `AllAnalyzers::Tag`.

**Review by rsmmr — CHANGES_REQUESTED** [review]

File: analyzer/main.zeek (hunk 1 above)
Comment (bbannier) [reply analyzer/main.zeek]: I should really be the other way around, https://github.com/zeek/zeek/blob/bf7b6e5ed09906e0f3345eb9b0c677f855c908a2/src/event.bif#L382-L383=.

File: analyzer/main.zeek [hunk 2]
```diff
@@ -289,7 +289,7 @@ function set_session(c: connection, message_id: int, opcode: LDAP::ProtocolOpcod
...
 @else
-event protocol_confirmation(c: connection, atype: AllAnalyzers::Tag, aid: count) {
+event protocol_confirmation(c: connection, atype: AllAnalyzer::Tag, aid: count) {
```
Comment (rsmmr) [inline analyzer/main.zeek]: The type *is* called `AllAnalyzers::Tag`.

Comment (bbannier) [reply analyzer/main.zeek]: Yikes, sorry for the back and forth. Looking at the 4.1.1 sources it seems just the name should have changed. Pushed an update.

File: analyzer/main.zeek [hunk 3]
```diff
@@ -289,7 +289,7 @@ function set_session(c: connection, message_id: int, opcode: LDAP::ProtocolOpcod
 @if (Version::at_least("4.2.0"))
 event analyzer_confirmation(c: connection, atype: Analyzer::Tag, aid: count) {
 @else
-event protocol_confirmation(c: connection, atype: AllAnalyzers::Tag, aid: count) {
+event protocol_confirmation(c: connection, atype: Analyzer::Tag, aid: count) {
 @endif
```
Comment (rsmmr) [inline analyzer/main.zeek]: I'm loosing track where we even started here, but it's still not quite right. I think this is what we want:

```
@if (Version::at_least("4.2.0"))
event analyzer_confirmation(c: connection, atype: AllAnalyzers::Tag, aid: count)
@else
event protocol_confirmation(c: connection, atype: Analyzer::Tag, aid: count) {
@endif
```

That said, I'm now wondering if we have a problem in Zeek (or whether I'm just confused at this point): Here are the prototypes in current `master`: 

```
# git grep '^event.*_confirmation' src/event.bif
src/event.bif:event analyzer_confirmation%(c: connection, atype: Analyzer::Tag, aid: count%);
src/event.bif:event protocol_confirmation%(c: connection, atype: AllAnalyzers::Tag, aid: count%) &deprecated="Remove in v5.1. Use analyzer_confirmation.";
```

At the same time, we're using the new event like this in `master`:

```
# git grep '^event.*_confirmation' scripts/
scripts/base/frameworks/dpd/main.zeek:event analyzer_confirmation(c: connection, atype: AllAnalyzers::Tag, aid: count) &priority=10
scripts/base/protocols/conn/inactivity.zeek:event analyzer_confirmation(c: connection, atype: AllAnalyzers::Tag, aid: count)
...
```

In 4.1 we have:

```
# git grep '^event.*_confirmation' scripts/
scripts/base/frameworks/dpd/main.zeek:event protocol_confirmation(c: connection, atype: Analyzer::Tag, aid: count) &priority=10
scripts/base/protocols/conn/inactivity.zeek:event protocol_confirmation(c: connection, atype: Analyzer::Tag, aid: count)
...
```

I think usage is right, but the prototypes look reversed. Paging @timwoj .

Comment (timwoj) [reply analyzer/main.zeek]: `analyzer_*` should be using `AllAnalyzers` everywhere. If there are spots where it isn't, then I probably screwed something up in that migration 😅

Comment (bbannier) [reply analyzer/main.zeek]: The way around this here seems to be to use `Analyzer::Tag` everywhere. I have filed zeek/zeek#2261 for any needed follow-up on the Zeek side.

File: analyzer/main.zeek (hunk 1 above)
Comment (rsmmr) [reply analyzer/main.zeek]: `AllAnalyzers`, no?

Comment (bbannier) [reply analyzer/main.zeek]: No, as with e.g., zeek-4.2.2 that would trigger

```
use of undeclared alternate prototype (event(c:connection; atype:error; aid:count;) and analyzer_confirmation
```

Comment (timwoj) [reply analyzer/main.zeek]: I'm not entirely sure why, but I changed that line to be `AllAnalyzers::Tag` and it worked just fine here. The plugin's tests passed and it installed. I didn't attempt to use it, but assuming the tests covered everything I don't know why it didn't work.

That said, I do think that `event.bif` should be changed to match the correct type, even if everywhere that uses that event seems to work.

Comment (bbannier) [reply analyzer/main.zeek]: I am not sure why `AllAnalyzers::Tag` didn't work for me locally, now it does. Sorry for the back and forth.

**Review by rsmmr — APPROVED** [review]

**Pull request merged**